tests/python/gpu/test_amp_init.py

```diff
@@ -0,0 +1,53 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import mxnet as mx
+from mxnet.gluon import nn
+from mxnet import amp
+import numpy as np
+import pytest
+
+
+@pytest.fixture
+def np_shape_array():
+    flags = mx.npx.is_np_shape(), mx.npx.is_np_array(), mx.npx.is_np_default_dtype()
+    mx.npx.set_np()
+    yield
+    mx.npx.set_np(*flags)
+
+
+@pytest.fixture(scope='module')
```

Comment:
Can we try to narrow the scope to `function`?

Reply:
It doesn't matter. The only reason I put scope='module' here is that amp.init() is called once per module. But even that doesn't matter much, since all subsequent calls would be a noop.
Regardless of what you declare in pytest amp.init() cannot be undone, at least with the current API.

Reply:
Should we add a utility to disable amp? If it is too much work to add quickly, you may add a separate pytest call in the `runtime_functions.sh` file for the amp test files.

Reply:
In fact, the amp in pytorch is implemented as a context manager.

Reply:
I agree it's a nifty API to enable/disable AMP with a context manager, but would definitely require a separate, non-trivial change.
For now, to unblock this PR, moved amp_init tests into a separate process.

Reply:
@mk-61 would you like to help create a tracking issue for that change? It sounds like something we should do before the stable 2.0 release

Reply:
@leezu - for now, I've just added an item here - https://github.com/apache/incubator-mxnet/issues/18896 - I keep it as a TODO list for AMP, of sorts. Still, can create a separate issue if you prefer to track it like this.
About adding it before the stable 2.0 release, well... To be clear - it's not just a matter of exposing something existing via a new API. Also, I'd like to clarify our plans about https://github.com/apache/incubator-mxnet/issues/18697 first, since it affects pretty much everything AMP-related.
When do we expect the stable 2.0 release? We still have some other, higher-priority items to fix.
Finally, my hope was that if a new API is going to be a strict superset of already existing one - i.e., we'll still keep amp.init() call, but also add a way to turn it off - may be it's acceptable to add it later? Or, to put it differently, may be it's less of an issue to add a new APIs, as long as we are not removing anything?

Reply:
> @leezu - for now, I've just added an item here - #18896 - I keep it as a TODO list for AMP, of sorts. Still, can create a separate issue if you prefer to track it like this.

That's fine. Thank you

>  About adding it before the stable 2.0 release, well... To be clear - it's not just a matter of exposing something existing via a new API. Also, I'd like to clarify our plans about #18697 first, since it affects pretty much everything AMP-related.

Using a graph pass may address the current issue of global state. A potential downside is that graph pass will only work with hybrid models.

> When do we expect the stable 2.0 release? We still have some other, higher-priority items to fix.

We'd like to create an Alpha release soon. There is no date for a stable release yet.

>  Finally, my hope was that if a new API is going to be a strict superset of already existing one - i.e., we'll still keep amp.init() call, but also add a way to turn it off - may be it's acceptable to add it later? Or, to put it differently, may be it's less of an issue to add a new APIs, as long as we are not removing anything?

It's fine to change the API in 2.0, if we feel it should be improved.